### PR TITLE
gke-topology-scheduler module implemented

### PR DESCRIPTION
**Pull request by sharabiani**
### Changes:

This module enables topology on a Google Kubernetes Engine cluster.
This is implemented based on sources and instructions explained [here](https://github.com/GoogleCloudPlatform/container-engine-accelerators/tree/master/gpudirect-tcpxo/topology-scheduler).

### CODE REVIEW

File: community/modules/compute/gke-topology-scheduler/main.tf [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module "kubectl_apply" {
+  source = "../../../../modules/management/kubectl-apply"
+
+  cluster_id = var.cluster_id
+  project_id = var.project_id
+
+  apply_manifests = [
```
Comment (ankitkinra) [inline community/modules/compute/gke-topology-scheduler/main.tf:21]: Is there a way to verify that node pool has compact policies applied ?

File: community/modules/compute/gke-topology-scheduler/manifests/schedule-daemon.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  selector:
+    matchLabels:
+      app: topology-scheduler
+  template:
+    metadata:
+      labels:
+        app: topology-scheduler
+    spec:
+      tolerations:
+      - key: "node-role.kubernetes.io/control-plane"
```
Comment (ankitkinra) [inline community/modules/compute/gke-topology-scheduler/manifests/schedule-daemon.yaml:33]: Should we add a toleration similar to kueue to allow it to land on system nodepool ? 

```
tolerations:
      - effect: NoSchedule
        key: components.gke.io/gke-managed-components
        operator: Equal
        value: "true"
```

File: community/modules/compute/gke-topology-scheduler/main.tf (hunk 1 above)
Comment (sharabiani) [reply community/modules/compute/gke-topology-scheduler/main.tf:21]: I'm not completely sure which nodepool?
We are not passing any node-pool module to this module.

Comment (ankitkinra) [reply community/modules/compute/gke-topology-scheduler/main.tf:21]: Ah okay , I thought topology scheduler only worked when we have nodes giving their placement information .  So if that is not found, topology scheduler is not helpful. But if that is not the case, feel free to disregard.

File: community/modules/compute/gke-topology-scheduler/manifests/schedule-daemon.yaml (hunk 2 above)
Comment (ankitkinra) [reply community/modules/compute/gke-topology-scheduler/manifests/schedule-daemon.yaml:33]: Which nodepool does Topology scheduler gets provisioned on ? By default our blueprints have two nodepools:
* system (Does not allow any thing to get provisioned until we add the toleration i was describing)
* GPU (we only want GPU workload to land here)

So did you add another nodepool to your blueprint for testing ?

File: community/modules/compute/gke-topology-scheduler/main.tf (hunk 1 above)
Comment (ankitkinra) [reply community/modules/compute/gke-topology-scheduler/main.tf:21]: Followed up offline, we will work on this toleration if it is required as part of blueprint that is using this topology scheduler

File: community/modules/compute/gke-topology-scheduler/manifests/schedule-daemon.yaml (hunk 2 above)
Comment (ankitkinra) [reply community/modules/compute/gke-topology-scheduler/manifests/schedule-daemon.yaml:33]: Followed offline, this testing will be done with full blueprint

**Review by ankitkinra — APPROVED** [review]

File: community/modules/compute/gke-topology-scheduler/main.tf (hunk 1 above)
Comment (sharabiani) [reply community/modules/compute/gke-topology-scheduler/main.tf:21]: Resolving this conversion. Will be tested with the xpk-a3 blueprints and modified if reuired.

**Pull request merged**